monitoring/src/listener/listener.rs

```diff
@@ -0,0 +1,54 @@
+use networking::p2p::network_channel::{NetworkChannelMsg, NetworkChannelRef, NetworkChannelTopic};
+use riker::actor::*;
+use std::fs::File;
+use crate::listener::records::Record;
+use std::io::Write;
+use log::*;
+
+type NetworkListenerRef = ActorRef<NetworkChannelMsg>;
+
+pub struct NetworkListener {
+    storage: File,
+    network_channel: NetworkChannelRef,
+}
+
+impl NetworkListener {
+    fn name() -> &'static str { "network-listener" }
+
+    fn new((storage_path, network_channel): (&'static str, NetworkChannelRef)) -> Self {
+        Self {
+            storage: File::open(storage_path).expect("Failed to open the file"),
```

Comment:
Return result instead of panicking.

Reply:
This is an out-dated file implementation, fixed in later commit.